Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/DownloadDiagnosisKeysTask.kt

```diff
@@ -174,6 +174,12 @@ class DownloadDiagnosisKeysTask @Inject constructor(
 
         val hasRecentDetection = nextForcedDetectionAt != null && now.isBefore(nextForcedDetectionAt)
 
+        val lastSuccesfulDetectionInFuture = nextForcedDetectionAt?.isBefore(now)
+        if(lastSuccesfulDetectionInFuture!=null&&lastSuccesfulDetectionInFuture){
+            Timber.tag(TAG).w("Last detection is more than 24h in the future. Assuming system date was wrong during last update.")
+            return false;
+        }
+
         return (hasRecentDetection && keySyncResult.newKeys.isEmpty()).also {
```

Comment:
Even if there was timetraveling, the `hasRecentDetectionAndNoNewFiles` would NOT abort, if new files have been downloaded as `keySyncResult.newKeys` should not be empty.